### PR TITLE
Test: Enable a new option to run test on ubuntu-next box.

### PR DESCRIPTION
**Pull request by eloycoto**
With this change, when a PR has the label ci/net-next the test will run
using net-next image instead of the base ubuntu image.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5921)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
hi @cilium/ci quick thing here: 

I updated the normal VM because the new kernel has the golang 1.11, so I need to update both. 
So this is the net-next + golang 1.11 support in the project. 

Regards

File: test/Vagrantfile [hunk 1]
```diff
@@ -9,13 +9,22 @@ $K8S_NODES = (ENV['K8S_NODES'] || "2").to_i
 $NFS = ENV['NFS']=="1"? true : false
 $SERVER_BOX = (ENV['SERVER_BOX'] || "cilium/ubuntu-dev")
 $SERVER_VERSION="120"
+$SERVER_VERSION= "119"
```
Comment (ianvernon) [inline test/Vagrantfile]: Why are there two "SEVER_VERSION" values here? Can you squash this with 7a2597bf41aabdbe4ed48b2391e4fe43a8ad7985 in case we need to bisect?

File: pkg/endpoint/endpoint_test.go [hunk 2]
```diff
@@ -424,63 +424,51 @@ func (s *EndpointSuite) TestProxyID(c *C) {
 }
 
 func TestEndpoint_GetK8sPodLabels(t *testing.T) {
-	type fields struct {
```
Comment (ianvernon) [inline pkg/endpoint/endpoint_test.go]: Why is this removed? `OpLabels` no longer has a mutex.

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by ianvernon** [comment]
@eloycoto sorry, for some reason I didn't see this was `wip`.

File: pkg/endpoint/endpoint_test.go (hunk 2 above)
Comment (eloycoto) [reply pkg/endpoint/endpoint_test.go]: Go vet complains about copying a sync.Mutex in there. This is why I deleted the mutex.

**Comment by eloycoto** [comment]
test-me-please

File: pkg/endpoint/endpoint_test.go [hunk 3]
```diff
@@ -424,63 +424,51 @@ func (s *EndpointSuite) TestProxyID(c *C) {
...
-		mutex    lock.RWMutex
-		OpLabels pkgLabels.OpLabels
-	}
 	tests := []struct {
 		name   string
-		fields fields
+		fields pkgLabels.OpLabels
```
Comment (aanm) [inline pkg/endpoint/endpoint_test.go]: keep the fields structure

File: pkg/endpoint/endpoint_test.go [hunk 4]
```diff
@@ -424,63 +424,51 @@ func (s *EndpointSuite) TestProxyID(c *C) {
 }
 
 func TestEndpoint_GetK8sPodLabels(t *testing.T) {
-	type fields struct {
-		mutex    lock.RWMutex
```
Comment (aanm) [inline pkg/endpoint/endpoint_test.go:428]: remove the mutex only

File: pkg/endpoint/endpoint_test.go [hunk 5]
```diff
@@ -424,63 +424,51 @@ func (s *EndpointSuite) TestProxyID(c *C) {
...
 			want: pkgLabels.Map2Labels(map[string]string{"foo": "bar"}, pkgLabels.LabelSourceK8s),
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			e := &Endpoint{
-				mutex:    tt.fields.mutex,
-				OpLabels: tt.fields.OpLabels,
+				mutex:    lock.RWMutex{},
+				OpLabels: tt.fields,
```
Comment (aanm) [inline pkg/endpoint/endpoint_test.go]: `tt.fields.OpLabels`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

File: pkg/endpoint/endpoint_test.go (hunk 2 above)
Comment (ianvernon) [reply pkg/endpoint/endpoint_test.go]: Why not change it to a pointer to a mutex instead of removing the mutex entirely?

**Review by aanm — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
Running tests without `ci/net-next` label to make sure that normal CI functionality works as intended with this change.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Tests passed without ci/net-next tag, this is good to merge.
